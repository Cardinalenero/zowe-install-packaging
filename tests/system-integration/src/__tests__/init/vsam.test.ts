/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

import { RemoteTestRunner } from '../RemoteTestRunner';
import { ZoweYaml } from '../ZoweYaml';

const testSuiteName = 'Dummy tests';
describe(testSuiteName, () => {
  beforeEach(() => {});

  it('run echo', async () => {
    const testRunner = new RemoteTestRunner();

    console.log('heres a log');
    const cfgYaml = ZoweYaml.basicZoweYaml();
    cfgYaml.java.home = '/ZOWE/node/J21.0_64/';
    const result = await testRunner.runRaw('echo "hi"');
    console.log(result);
    console.log(JSON.stringify(result));
    expect(result.rc).toBe(0); // 100 is expected...
    expect(result.stdout).not.toBeNull();
    expect(result.stdout).toMatchSnapshot();

    const result2 = await testRunner.runRaw('cat /u/zowead3/.profile');
    console.log(result2);
    console.log(JSON.stringify(result2));
    expect(result2.rc).toBe(0); // 100 is expected...
    expect(result2.stdout).not.toBeNull();
    expect(result2.stdout).toMatchSnapshot();
  });

  it('a test', async () => {
    const testRunner = new RemoteTestRunner();
    console.log('heres a log');
    const cfgYaml = ZoweYaml.basicZoweYaml();
    cfgYaml.java.home = '/ZOWE/node/J21.0_64/';
    const result = await testRunner.runTest(cfgYaml, '--help');
    console.log(result);
    console.log(JSON.stringify(result));
    expect(result.rc).toBe(100); // 100 is expected...
    expect(result.stdout).not.toBeNull();
    expect(result.stdout).toMatchSnapshot();
  });
});
