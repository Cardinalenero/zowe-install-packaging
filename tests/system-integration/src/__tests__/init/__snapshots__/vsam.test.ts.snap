// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dummy tests a test 1`] = `
"
$ zwe 

------------------
Synopsis
    zwe  [sub-command [sub-command]...] [parameter [parameter]...]

------------------
Description
    A command line utility helps you managing Zowe instance.
    
    You can issue --help or -h to find information for all commands it supports.

------------------
Global parameters
    --help|-h  (optional)
        Display this help. 
    --debug|--verbose|-v  (optional)
        Enable verbose mode. 
    --trace|-vv  (optional)
        Enable trace level debug mode. 
    --silent|-s  (optional)
        Do not display messages to standard output. 
    --log-dir|--log|-l string (optional)
        Write logs to this directory. 
    --config|-c string (optional)
        Path to Zowe configuration zowe.yaml file. 
    --configmgr  (optional)
        Enable use of configmgr capabilities. 
------------------
Available sub-command(s)
    - certificate
    - components
    - config
    - diagnose
    - init
    - install
    - internal
    - migrate
    - sample
    - start
    - stop
    - support
    - version

------------------
Example(s)
    zwe install -h
    
    zwe init --allow-overwrite --config /path/to/zowe.yaml
    
    zwe start -c /path/to/zowe.yaml"
`;

exports[`Dummy tests run echo 1`] = `
"
$ hi"
`;

exports[`Dummy tests run echo 2`] = `
"
$ export TERM=xterm

# python3
export PYTHON_HOME=/ZOWE/node/python3/usr/lpp/IBM/cyp/v3r12/pyz

# java 8
export JAVA_HOME=/ZOWE/node/J8.0_64

# these variables are required by v8, v12
export _BPXK_AUTOCVT=ON
export _CEE_RUNOPTS="FILETAG(AUTOCVT,AUTOTAG) POSIX(ON)"

# node v14.15.1
#export NODE_HOME=/ZOWE/node/node-v14.21.3-os390-s390x

# node v16.20.1
export NODE_HOME=/ZOWE/node/node-v16.20.2-os390-s390x

# node v18.x
# export NODE_HOME=/ZOWE/node/node-v18.16.0

export PATH=/bin:.:/ZOWE/bin:$NODE_HOME/bin:$JAVA_HOME/bin:$PYTHON_HOME/bin"
`;
